raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -509,7 +509,8 @@ private VoteResponseData handleVoteRequest(
         VoteRequestData request = (VoteRequestData) requestMetadata.data;
 
         if (!hasValidTopicPartition(request, log.topicPartition())) {
-            return VoteRequest.getPartitionLevelErrorResponse(request, Errors.UNKNOWN_TOPIC_OR_PARTITION);
+            // Until we support multi-raft, we treat topic partition mismatches as invalid requests
+            return new VoteResponseData().setErrorCode(Errors.INVALID_REQUEST.code());
```

Comment:
@abbccdda After thinking about it, I decided to change this handling. It's more effort to try and handle it correctly and probably not worth the effort since it is not supported anyway. For example, it's possible that one of the included partitions _does_ match the metadata partition correctly. Rather than trying to detect that, seems like it's simpler to just reject the request.

Reply:
Hmm, but is it worth adding the partial handling logic? I am thinking not. There's not a legitimate scenario where we would see this until we have multi-raft support.